### PR TITLE
small fixes for mlir backend

### PR DESCRIPTION
**Pull request by ngimel**
Fixes for skipped tests with mlir triton backend (will unskip once #94249 lands)

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Review by atalman — APPROVED** [review]

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -129,6 +129,10 @@ class TritonOverrides(OpOverrides):
     def to_dtype(x, dtype: torch.dtype):
         if dtype == torch.bool:
             return f"({x} != 0)"
+        elif dtype == torch.uint8:
+            # to work around llvm uint conversion semantics
+            # that produces 0's for negative values
+            return f"{x}.to(tl.int8).to(tl.uint8)"
```
Comment (anmyachev) [inline torch/_inductor/codegen/triton.py:135]: HI @ngimel! 

I encountered a problem that this code does not work correctly on the XPU backend, after a little debugging it turned out that for `to(tl.int8).to(tl.uint8)` only conversion to i8 is generated, and there is no conversion to ui8 (for all backends). So PyTorch rely on buggy code and the model works only by a lucky chance. For more context - https://github.com/triton-lang/triton/issues/7502.

Do you have time to roll back this code and see how the model will work with the current Triton? Unfortunately I don't have cuda environment.

Comment (davidberard98) [reply torch/_inductor/codegen/triton.py:135]: @anmyachev would you be able to put up a PR and we can see the results in CI?

Comment (anmyachev) [reply torch/_inductor/codegen/triton.py:135]: Hi @davidberard98! Ok, I'll do it.

**Pull request closed without merging**